modules/gltf/gltf_document.h

```diff
@@ -86,6 +87,8 @@ class GLTFDocument : public Resource {
 	static void unregister_gltf_document_extension(Ref<GLTFDocumentExtension> p_extension);
 	static void unregister_all_gltf_document_extensions();
 
+	void set_compatibility_version(int p_version);
```

Comment:
To be completely honest, I would not call it `compatibility_version` but simply some flag such as "use_compatibility_naming". Because you may want to have new features in the GLTF2 supported at some point while users would not want to miss on those ones, while keeping this compatibility setting on.

Reply:
`use_compatibility_naming` makes it sound like a boolean, but we might want future changes to naming which would lead to needing yet another boolean. In fact, aaron has some open PRs which might also impact node naming but were put on hold due to this exact problem.

on the other hand, if we can solve the gltf node id problem before then then name compatibility might not be as big a problem in the future and a boolean would be sufficient

Reply:
if you want to do this its fine, but then I think it should be called naming_scheme_version or something like that, not be a general GLTF version thing.

Reply:
I really don't understand why we complicate it so much. We have one user-facing interface: the importer. Whatever it does, it produces a different output in Godot 4.2 compared to Godot 4.1. That's all that really matters, one version increment.

This only needs to exist to allow a smooth transition to 4.2. If users want to reimport their assets in 4.2 and use new features from 4.2+, then it's fine if we fully upgrade them to use the most up to date format. Users will have to update their affected scenes, but it will be clear there and then, because they are manually reimporting those assets.

We don't need to support individual changes so granularly, it's not going to scale for years to come. And there is no reason why someone would want to have naming scheme 4 with something else version 2 with something else 10 in Godot 4.8. It should only exist to support legacy settings during the transition, and then it should be up to user to accept the need to migrate or not use any new features.

And we don't have to think about all the permutations.

Reply:
I prefer `compatibility_version`, because what matters here is like Yuri said, providing compatibility for older Godot versions. There is no reason to use the names from 4.1 and the "something else" from 4.8 together, so I don't think it's a benefit to have a separate setting for names vs other things.

Reply:
@aaronfranke the problem in this case, is that conceptually, this only really matters for names, so I think doing futurology on what may change that may require tweaking this again, and that it does in a way where you still care about the names to me is going into speculative territory. This is why I would keep this option strictly for naming, which is what we know may change in the future and should not affect anything else, hence why I suggest `naming_scheme_version` instead and do the dedicated solution to the problem you know you have instead.